shared/types/models.go

```diff
@@ -251,35 +251,30 @@ func (train *Train) SendCommitCountMetrics() {
 	datadog.Count("commit.no_verify", noVerifyCommits, train.DatadogTags())
 }
 
-func DoesCommitNeedTicket(commit *Commit, commitsOnTickets map[string]struct{}) bool {
+func DoesCommitNeedTicket(commit *Commit, commitsOnTickets map[string]struct{}, noStagingVerify bool) bool {
 	_, found := commitsOnTickets[commit.SHA]
-	// Exclude commits with tickets and commits marked for no verification.
-	if found || commit.IsNoVerify() {
-		return false
-	}
-	// Exclude users in the no staging pilot program, unless they manually
-	// requested staging.
-	if commit.IsNoStagingVerification() && !commit.IsNeedsStaging() {
-		return false
+	// Exclude commits that already have tickets. Include Staging tickets.
+	if !found && commit.IsNeedsStaging(noStagingVerify) {
+		return true
 	}
-	return true
+	return false
 }
 
 // Should this commit trigger slack notifications to its author regarding staging.
-func (commit *Commit) DoesCommitNeedStagingNotification() bool {
-	return !commit.IsNoStagingVerification() || commit.IsNeedsStaging()
+func (commit *Commit) DoesCommitNeedStagingNotification(noStagingVerify bool) bool {
+	return !commit.IsNoStagingVerification(noStagingVerify) || commit.IsNeedsStaging(settings.NoStagingVerification)
 }
 
 func (commit *Commit) IsNoVerify() bool {
 	return strings.Contains(commit.Message, "[no-verify]")
 }
 
-func (commit *Commit) IsNeedsStaging() bool {
-	return strings.Contains(commit.Message, "[needs-staging]")
+func (commit *Commit) IsNeedsStaging(noStagingVerify bool) bool {
```

Comment:
I could be convinced otherwise, but this feels like some special case (in the sense that we're only allowing [staging-verify]. Would it be better to have a method that takes all the commit strings, and returns the level of verification necessary? It should be easier to test, and you wouldn't need to pass `bool`s  around.

If that ends up being a major refactor we can shelve for later - what do you think?

Reply:
I was hoping to have a discussion about removing the NoStagingVerify flag completely from Conductor (as well as the pilot program users) and having users rely just on the `[needs-staging]` sigil. That would be a more major refactor that would nullify some of this passing of bools around. In the meantime, I'll take another look at this flow and at least limit the number of methods used; I think that should be possible